src/RestSharp/RestClient.Async.cs

```diff
@@ -95,38 +96,67 @@ public partial class RestClient {
 
         var ct = cts.Token;
 
+        
+        HttpResponseMessage? responseMessage;
+        // Make sure we have a cookie container if not provided in the request
+        CookieContainer cookieContainer = request.CookieContainer ??= new CookieContainer();
+        
+        var headers = new RequestHeaders()
+            .AddHeaders(request.Parameters)
+            .AddHeaders(DefaultParameters)
+            .AddAcceptHeader(AcceptedContentTypes)
+            .AddCookieHeaders(url, cookieContainer)
+            .AddCookieHeaders(url, Options.CookieContainer);
+
+        message.AddHeaders(headers);
+        if (request.OnBeforeRequest != null) await request.OnBeforeRequest(message).ConfigureAwait(false);
+        await OnBeforeRequest(message).ConfigureAwait(false);
+        
         try {
-            // Make sure we have a cookie container if not provided in the request
-            var cookieContainer = request.CookieContainer ??= new CookieContainer();
-
-            var headers = new RequestHeaders()
-                .AddHeaders(request.Parameters)
-                .AddHeaders(DefaultParameters)
-                .AddAcceptHeader(AcceptedContentTypes)
-                .AddCookieHeaders(url, cookieContainer)
-                .AddCookieHeaders(url, Options.CookieContainer);
-
-            message.AddHeaders(headers);
-
-            if (request.OnBeforeRequest != null) await request.OnBeforeRequest(message).ConfigureAwait(false);
-
-            var responseMessage = await HttpClient.SendAsync(message, request.CompletionOption, ct).ConfigureAwait(false);
-
+            responseMessage = await HttpClient.SendAsync(message, request.CompletionOption, ct).ConfigureAwait(false);
```

Comment:
What was the reason to move the declaration of `responseMessage` to line 100?

Reply:
The reason, was that I handle the response inside the interceptor and prevent the tryCatch-Block from, catching exceptions thrown inside the interceptor.
For example I use an interceptor, to preprocess ErrorMessage to convert them back into Exceptions.